app/views/users/index.html.erb

```diff
@@ -1,5 +1,10 @@
 <h2>People: </h2>
 <br>
 <% @users.each do |user| %>
-  <h4> <%= link_to user.name, user_path(user.id) %> </h4>
-<% end %>
\ No newline at end of file
+  <% if user != current_user %>
+    <h4><%= link_to user.name, user_path(user.id) %></h4>
+    <% if !FriendRequest.exists?(user: current_user, friend: user) && current_user.pending_friends.include?(user)%>
```

Comment:
- [ ] So it turns that you already have the sending request functionality, but:

- `current_user.pending_friends.include?(user)` will not work as expected, because as per our association:
```
has_many :pending_friends, through: :friend_requests, source: :friend
```
`sender.pending_friends` will return `receiver`, but `receiver.pending_friends` will not return `sender`. It will always return the users that have received the request because the `source: :friend`. We can change it to `source: :user` but It can break another functionality.